core/pkg/eval/string_comparison_evaluation.go

```diff
@@ -0,0 +1,110 @@
+package eval
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+)
+
+// startsWithEvaluation checks if the given property starts with a certain prefix.
+// It returns 'true', if the value of the given property starts with the prefix, 'false' if not.
+// As an example, it can be used in the following way inside an 'if' evaluation:
+//
+//	{
+//	  "if": [
+//			{
+//				"starts_with": [{"var": "email"}, "user@faas"]
+//			},
+//			"red", null
+//			]
+//	}
+//
+// This rule can be applied to the following data object, where the evaluation will resolve to 'true':
+//
+// { "email": "user@faas.com" }
+//
+// Note that the 'starts_with'  evaluation rule must contain exactly two items, which both resolve to a
+// string value
+func (je *JSONEvaluator) startsWithEvaluation(values, _ interface{}) interface{} {
+	propertyValue, target, err := parseStringComparisonEvaluationData(values)
+	if err != nil {
+		je.Logger.Error(fmt.Sprintf("parse starts_with evaluation data: %v", err))
+		return nil
+	}
+	return strings.HasPrefix(propertyValue, target)
+}
+
+// endsWithEvaluation checks if the given property starts with a certain prefix.
+// It returns 'true', if the value of the given property starts with the prefix, 'false' if not.
+// As an example, it can be used in the following way inside an 'if' evaluation:
+//
+//	{
+//	  "if": [
+//			{
+//				"ends_with": [{"var": "email"}, "faas.com"]
+//			},
+//			"red", null
+//			]
+//	}
+//
+// This rule can be applied to the following data object, where the evaluation will resolve to 'true':
+//
+// { "email": "user@faas.com" }
+//
+// Note that the 'ends_with'  evaluation rule must contain exactly two items, which both resolve to a
+// string value
+func (je *JSONEvaluator) endsWithEvaluation(values, _ interface{}) interface{} {
+	propertyValue, target, err := parseStringComparisonEvaluationData(values)
+	if err != nil {
+		je.Logger.Error(fmt.Sprintf("parse ends_with evaluation data: %v", err))
+		return nil
+	}
+	return strings.HasSuffix(propertyValue, target)
+}
+
+// parseStringComparisonEvaluationData tries to parse the input for the starts_with/ends_with evaluation.
+// this evaluator requires an array containing exactly two strings.
+// Note that, when used with jsonLogic, those two items can also have been objects in the original 'values' object,
+// which have been resolved to string values by jsonLogic before this function is called.
+// As an example, the following values object:
+//
+//	{
+//	  "if": [
+//			{
+//				"starts_with": [{"var": "email"}, "user@faas"]
+//			},
+//			"red", null
+//			]
+//	}
+//
+// with the following data object:
+//
+// { "email": "user@faas.com" }
+//
+// will have been resolved to
+//
+// ["user@faas.com", "user@faas"]
+//
+// at the time this function is reached.
+func parseStringComparisonEvaluationData(values interface{}) (string, string, error) {
+	parsed, ok := values.([]interface{})
+	if !ok {
+		return "", "", errors.New("starts_with evaluation is not an array")
```

Comment:
I don't have strong opinion how the log line looks like but I think we should provide info about "ends_with". 
If I start playing with the targeting rule using "ends_with" making a mistake and in the log I see "starts_with", I would be completely lost 😅 
```suggestion
		return "", "", errors.New("[start/end]s_with evaluation is not an array")
```

Reply:
agree, will adapt that :)